lib/Cake/Utility/File.php

```diff
@@ -89,7 +89,7 @@ public function __construct($path, $create = false, $mode = 0755) {
 			$this->name = basename($path);
 		}
 		$this->pwd();
-		$create && !$this->exists() && $this->safe($path) && $this->create();
+		$create && !$this->exists() && $this->safe($path) && $this->create($mode);
```

Comment:
The `$mode` there is supposed to be (awful api...) [the mode of the _containing folder](https://github.com/cakephp/cakephp/blob/master/lib/Cake/Utility/File.php#L83).